src/libYARP_OS/src/Time.cpp

```diff
@@ -113,34 +92,183 @@ void Time::yield() {
 }
 
 
-void Time::useSystemClock() {
-    lock();
-    removeClock();
-    unlock();
+bool Time::useSystemClock() {
+    if(!isSystemClock())
+    {
+        lock();
+
+        Clock *old_pclock = pclock;
+        bool old_clock_owned = clock_owned;
+
+        pclock = new SystemClock();
```

Comment:
Meeting minutes: 
I know, this implementation do takes this into account. All functions `Now()` and `Delay()` call the static methods:
```
double Time::now() {
     if(isSystemClock())
         return SystemClock::nowSystem();
 ...
```

The system clock is instantiate anyway but if everything is working correctly, the interface is never used.
I did it because the the check
`if pointer == NULL` 
then everything is fine go on, looked a bit weird to me.

Now an instance of clock (whichever version) is always on and no null pointers are wandering around. An enum is been created to verify the clock type (system, network or custom)